internal/translation/deployment/conversion.go

```diff
@@ -474,8 +474,8 @@ func labelsFromAtlas(cLabels []admin.ComponentLabel) []common.LabelSpec {
 }
 
 func replicationSpecFromAtlas(replicationSpecs []admin.ReplicationSpec) []*akov2.AdvancedReplicationSpec {
-	hSpecOrDefault := func(spec admin.HardwareSpec) *akov2.Specs {
-		if spec.GetNodeCount() == 0 {
+	hSpecOrDefault := func(spec admin.HardwareSpec, providerName string) *akov2.Specs {
+		if spec.GetNodeCount() == 0 && providerName != string(provider.ProviderTenant) {
```

Comment:
Why was this also necessary?

Reply:
during comparison, we were turning node spec with nodeCount 0 to nil. As the shared cluster has only instance size defined (nodeCount will be always 3 and the API omits the value :man_shrugging: ), we need to consider this when preparing it to compare.

Reply:
sorry, but what does the provider name have to do with the node count or instance size? I am missing that detail

Reply:
we identify shared cluster by provider name (TENANT)